### PR TITLE
Store Manager and Store Admin can manage item information

### PR DESCRIPTION
**Pull request by marshallhouston**
#### Pivotal URL: 
https://www.pivotaltracker.com/story/show/153614689
https://www.pivotaltracker.com/story/show/153568563

#### What does this PR do?
This PR only shows items associated with specific stores that a user is either a store_manager or a store_admin. A platform admin will still see all items.

If a store_manager/admin has multiple stores, all of the items will show up, so I added a store name to each item.

It maintains the original functionality to change item information.

#### Where should the reviewer start?
app/controllers/admin/items_controller.rb to see the method that scopes the items to a specific user.
item.rb to see the method that limits the collection of items.

#### How should this be manually tested?
log in as a store_admin or a store_manager and see if you can change the status of an order.

#### What are the relevant story numbers?
153614689
153568563
#### Questions:
  - Do Migrations Need to be ran?
No
  - Do Environment Variables need to be set?
No
  - Any other deploy steps?
No

### CODE REVIEW

File: spec/features/store_admin/store_admin_can_change_item_information_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+feature "As a store manager" do
+  let(:store_admin) { create(:user) }
+  let(:store_admin_role) { create(:store_admin) }
+  let(:store) { create(:store, status: "active") }
+  let(:store_item) { create(:item, store: store) }
+
+  scenario "I can change item information for a store that I manage" do
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(store_admin)
```
Comment (izaxveta) [inline spec/features/store_admin/store_admin_can_change_item_information_spec.rb:10]: If this is applied more than once it can be pulled out into a before :each method.

File: spec/features/store_admin/store_admin_can_change_item_information_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    fill_in "item[description]", with: "Improve your vertical leap"
+    fill_in "item[price]", with: "32.52"
+    page.attach_file("item[image]", testing_image)
+    click_on "Update"
+
+    expect(page).to have_content("Learning to Jump")
+    expect(page).to have_content("$32.52")
+  end
+
+  scenario "I don't see an item if it is not in a store I manage" do
```
Comment (izaxveta) [inline spec/features/store_admin/store_admin_can_change_item_information_spec.rb:29]: Good edge case test

**Comment by izaxveta** [comment]
Great edge case testing, code is clean and concise!
